resources/views/pages-legacy/download.blade.php

```diff
@@ -18,81 +16,88 @@
     <h2 class="mb-6">Download a supported emulator</h2>
 
     @foreach ($emulators as $emulator)
-        <h2 class="longheader" id="<?= mb_strtolower($emulator['handle'] ?? null) ?>">
-            <a href="#<?= mb_strtolower($emulator['handle'] ?? null) ?>"><?= $emulator['handle'] ?? null ?></a>
-            @if ($emulator['handle'] != $emulator['name'])
-                <small>(<?= $emulator['name'] ?? null ?>)</small>
+        @php
+            $systems = $emulator->systems()->active()->orderBy('Name')->pluck('Name')->toArray();
```

Comment:
There seems to be an N+1 query problem here. It's not debilitating by any means, but could be worth adding an eager load somewhere while we're already in here.

<img width="1448" alt="Screenshot 2024-09-11 at 9 40 53 PM" src="https://github.com/user-attachments/assets/afeb3ccc-b04c-4ea6-a366-16157e72e569">

Reply:
Can you eagerly load a HasMany relationship? I thought that only works for a BelongsTo relationship.

Reply:
Yep, you can eagerly load any relationship.

```php
$emulators = Emulator::active()
    ->orderBy('name')
    ->with(['latestRelease', 'systems' => function ($query) {
        $query->active()->orderBy('Name');
    }])
    ->get();
```

```php
@php
    $systems = $emulator->systems->pluck('Name')->toArray();
    if (empty($systems)) {
        continue;
    } else {
        // just some debugging output
        dump($systems);
    }
@endphp
```

Reply:
I'm not seeing any difference.

Before:
![image](https://github.com/user-attachments/assets/c8143a10-b938-479a-98c9-6b3c0f636e5e)

With your changes:
![image](https://github.com/user-attachments/assets/ae9ab578-f1ee-41d0-8bf8-09939e0fb19d)

It's just queries for the systems sooner.

It does seem to use one less query, but generates twice as many models.

Reply:
I thought I had done that, but where you have

```
 $systems = $emulator->systems->pluck('Name')->toArray();
```

I had

```
 $systems = $emulator->systems()->pluck('Name')->toArray();
```

Removing those extra parenthesis did drop the number of queries, but the overall time was unchanged (still ~150ms)

Reply:
Yep, not debilitating by any means as the queries themselves are very fast, there's just no reason in this case to run them in a waterfall.

The `->relations` vs `->relations()` thing trips me up often as well.

Reply:
Oh, that's page load time. Query time dropped from 6.6ms to 5.3ms. 

Changes applied and committed.